### DISCUSSION TITLE
Battery with Solid Electrolyte and Li-Me-Anode

### DISCUSSION THREAD

**Discussion by Nata2340** [pybamm-team/PyBaMM#2549] [General]
Hello, I'm working for the first time with PyBamm and unfortunately I do not know so well yet. I would like to model a battery with the following properties:

Cathode: Li active material with particle size distribution
Electrolyte: solid, no separator
Anode: Li-Metal

I had two ideas for this, which I came up with using the notebooks.

1. preset DFN model with changed options
2. base model: half cell

First try: Here I used the following formulation:

**options = {"working electrode": "positive", "particle size": "distribution"} 
model = pybamm.lithium_ion.DFN(options=options)
model.submodels["lithium plating"] = pybamm.lithium_plating.NoPlating(model.param, model.options)
model.submodels["sei"] = pybamm.sei.NoSEI(model.param, model.options, cracks=False)**

However, the error always appears:
_" KeyError: KeyError("'Typical plated lithium concentration [mol.m-3]' not found. Best matches are ['Typical electrolyte concentration [mol.m-3]', 'Bulk solvent concentration [mol.m-3]', 'Lithium interstitial reference concentration [mol.m-3]']")"_

I read from another discussion that a lithium plating parameter set is needed for this. I think at this point I don't understand how to represent a Li-Me anode without plating. Or is this not possible?

Second try 2: Here I tried to display only the active material electrode without separator and followed the notebook for Creating Models-Half cell. For the potential I left out only the separator part and adapted further in the following. 

**phi = pybamm.Variable("Positive electrode potential [V]", domain="positive electrode")
phi_e_p = pybamm.Variable("Positive electrolyte potential [V]", domain="positive electrode")**

I hope it is correct, but the system could calculate something. 
But since I have to consider a concentration gradient in the electrolyte, I have added the equations for this to the half-cell model. However, many errors occur because of this, e.g. 

_"ModelError: model is underdetermined (too many variables)"_

Is there an easier way to create the desired model? I would be very happy about it. Many thanks in advance!

**Comment by PaulKranefeld** [comment]
Hello, 
unfortunately I can't give you an exact solution for your problem but I've been working on a very similar problem for a couple of weeks and first tried to adapt existing models as well. This turned out to be complicated so far. From my point of view it is currently not possible to adapt the existing models simply by making small changes. Correct me if I am wrong, I am also new to Pybamm. So I started to design my own model.  Here I started with the basics and then slowly added individual aspects. What I have so far looks promising. Basically, the half cell model was a good starting point for me.
